### PR TITLE
Validate the send_msg template only when not empty

### PR DESCRIPTION
**Pull request by norkans7**
Fix https://textit.sentry.io/issues/5293847588/

### CODE REVIEW

File: flows/actions/testdata/send_msg.json [hunk 1]
```diff
@@ -21,6 +21,31 @@
         },
         "read_error": "field 'topic' is not a valid message topic"
     },
+    {
```
Comment (norkans7) [inline flows/actions/testdata/send_msg.json:24]: this tests are just to make sure the validation works but no easy way to test the validation with omitempty

File: flows/actions/send_msg.go [hunk 2]
```diff
@@ -47,7 +47,7 @@ type SendMsgAction struct {
 	createMsgAction
 
 	AllURNs           bool                      `json:"all_urns,omitempty"`
-	Template          *assets.TemplateReference `json:"template,omitempty"`
+	Template          *assets.TemplateReference `json:"template,omitempty" validate:"omitempty"`
```
Comment (rowanseymour) [inline flows/actions/send_msg.go:50]: if this was necessary... every single `send_msg` action without a template would be failing .. no?

Comment (norkans7) [reply flows/actions/send_msg.go:50]: Yes, I think those that are failing are those with a previous version set before they migrate to the latest version

Comment (rowanseymour) [reply flows/actions/send_msg.go:50]: And why are we only seeing this problem for simulation?

Comment (norkans7) [reply flows/actions/send_msg.go:50]: mailroom will use the flow definition passed from the simulation request and that is the definition before the migrations I think

**Pull request closed without merging**